p2p/simulations/network.go

```diff
@@ -571,3 +576,69 @@ func (self *Network) Shutdown() {
 		}
 	}
 }
+
+func ConnLabel(source, target *adapters.NodeId) string {
+	var first, second *adapters.NodeId
+	if bytes.Compare(source.Bytes(), target.Bytes()) > 0 {
+		first = target
+		second = source
+	} else {
+		first = source
+		second = target
+	}
+	return fmt.Sprintf("%v-%v", first, second)
+}
+
+type Snapshot struct {
+	Nodes []NodeSnapshot `json:"nodes,omitempty"`
+	Conns []Conn         `json:"conns,omitempty"`
+}
+
+type NodeSnapshot struct {
+	Node
+
+	Snapshot []byte `json:"snapshot,omitempty"`
+}
+
+func (self *Network) Snapshot() (*Snapshot, error) {
+	self.lock.Lock()
+	defer self.lock.Unlock()
+	snap := &Snapshot{
+		Nodes: make([]NodeSnapshot, len(self.Nodes)),
+		Conns: make([]Conn, len(self.Conns)),
+	}
+	for i, node := range self.Nodes {
+		snapshot, err := node.Snapshot()
```

Comment:
I have modified this to only snapshot running nodes: https://github.com/ethersphere/go-ethereum/blob/d7eb6b5a70d4f49887ce982068007821dad8b407/p2p/simulations/network.go#L615-L625